internal/state/state_test.go

```
@@ -0,0 +1,1297 @@
+// A full or read-only $STATE_DIR must surface as an error (mapped to exit
+// 5 by the CLI, S.6), not a silently-discarded writeAtomic failure that
+// reports success while nothing was actually persisted.
+func TestProcess_PropagatesWriteFailures(t *testing.T) {
+	cfg := testConfig(t, time.Unix(1000, 0))
+	s := newStore(t, cfg)
+
+	alertDir := filepath.Join(cfg.StateDir, "active-alerts")
+	if err := os.Chmod(alertDir, 0o500); err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { os.Chmod(alertDir, 0o700) }) // let t.TempDir() clean up
```

golangci-lint (2.12.2)

[error] Error return value of `os.Chmod` is not checked

(errcheck)